pkg/controller/ingress/ingress_controller.go

```diff
@@ -184,13 +188,13 @@ func (r *ReconcileIngress) fetchNetworkingInfo(instance *extensionsv1beta1.Ingre
 		nodeInstanceIds = append(nodeInstanceIds, node.Spec.ProviderID[strings.LastIndex(node.Spec.ProviderID, "/")+1:])
 	}
 
-	r.log.Info("getting vpcID, securityGroups, subnetIds for worker nodes")
-	vpcIDs, subnetIds, securityGroups, err := network.GetNetworkInfoForEC2Instances(r.ec2Svc, nodeInstanceIds)
+	r.log.Info("getting vpcID, securityGroups, subnetIds, asgNames for worker nodes")
+	vpcIDs, subnetIds, securityGroups, asgNames, err := network.GetNetworkInfoForEC2Instances(r.ec2Svc, r.autoscalingSvc, nodeInstanceIds)
```

Comment:
Nitpick: To improve readability we can return a single *network.Network object in place of these four lists.

Reply:
Not sure, I didn't try much to refactor old code, but we can improve on it later on.